packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -602,6 +602,10 @@ class TextPainter {
     return value & 0xF800 == 0xD800;
   }
 
```

Comment:
code comment for what the scenario is and what this does

Reply:
Hi, I'm finishing my flutter app, but some bugs are preventing me to release it.
in textfield if you have a persian numbers almost it always buggy, you know in persian numbers are from left to right but characters are from right to left, when users input for example mobile number, it's getting write in the opposite direction and it is just happening in ANDRIOD, in iOS it is smooth and fine